### PR TITLE
Skip node station when determining which RL branches are affected by alert

### PR DESCRIPTION
**Pull request by hannahpurcell**
**Asana task**: [Bug: GL text can appear on RL pre-fare alerts](https://app.asana.com/0/1185117109217413/1206815194782600)

It's a little tricky figuring out which RL destination sign to show when the alert affects both the RL trunk and branch. So here are the cases we need to support
- issue is downstream, JFK - Ashmont and no Braintree stations included: should say riders to Ashmont
- issue is downstream, Broadway - Ashmont: should say riders to both branches
- issue is downstream, JFK - Braintree and no Ashmont stations included: should say riders to Braintree

Omitting the branching node (JFK) helps us decide whether the trunk/branch is affected

- [x] Tests added?

### CODE REVIEW

**Review by cmaddox5 — APPROVED** [review]
Looks good! Just have a non-blocking suggestion.

File: lib/screens/v2/widget_instance/reconstructed_alert.ex [hunk 1]
```diff
@@ -228,6 +229,22 @@ defmodule Screens.V2.WidgetInstance.ReconstructedAlert do
...
+      |> elem(0)
+
+    if {direction_id, "Red"} in list_of_directions_and_routes or
+         ({direction_id, "Red-Ashmont"} in list_of_directions_and_routes and
+            {direction_id, "Red-Braintree"} in list_of_directions_and_routes) do
+      {direction_id, "Red"}
+    else
+      {direction_id, "Green-trunk"}
+    end
+  end
```
[the remark this answers is not in the document]
Comment (hannahpurcell) [reply lib/screens/v2/widget_instance/reconstructed_alert.ex:246]: The suggested change doesn't contain the same logic as the original, but that tells me I should definitely simplify the if/else then! I'll do that

[the remark this answers is not in the document]
Comment (hannahpurcell) [reply lib/screens/v2/widget_instance/reconstructed_alert.ex:246]: OH I misunderstood. That could work! Thanks!

[the remark this answers is not in the document]
Comment (hannahpurcell) [reply lib/screens/v2/widget_instance/reconstructed_alert.ex:246]: Nope, you're readability was great! I thought it skipped some cases, but it covers everything. I was way overcomplicating things.

[the remark this answers is not in the document]
Comment (hannahpurcell) [reply lib/screens/v2/widget_instance/reconstructed_alert.ex:246]: It just includes a little typo, so I'm going to edit your suggested code so I can directly apply it

**Pull request merged**